### PR TITLE
LCOW: Mount to short container paths to avoid command-line length limit

### LINKED ISSUE #36764: Multistage COPY operations in LCOW fails if the source image has more than 56 commands.
**Description**

While building an image with over 50 COPY operations from other images docker gives following error:
`COPY failed: Forbidden path outside the build context: test ()`

Docker does not write the full error. I have changed the codes to print the actual error message. It is something like this:
```
failed to copy files: svm.runProcess: command mount -t overlay overlay -olowerdir=/tmp/ac584b2bf23b1850f95d26f5f9fe788fdb38f024f4b588706f9adab142745ebb:/tmp/b058ef8dc02f3b0704255e7c07b3d26f129152c614ae3f10dcdb40129aef4cee:/tmp/9856e8860ee641e2b09a48d0acccad741f8a8b54d29323c275c68a70aee2db66:/tmp/5e90b26e0577694a8437a6ebab87a95d93eea574a1b93c24ee4a5127d83c83c7:/tmp/d8225f78bcc6df46b3d0c497bb68562333595a9a656e117d675616e31129ffa8:/tmp/8308a98b961813400977fc3d77f7dd4a3e2f20208a5318f65771fc4357c26b53:/tmp/697752da9f17759760387aab800cf3ab464cc265d48f65dfd9c44b4b38dbf391:/tmp/90e1f09e3cb8a8e439ef666d35c66aa9e1dabcb47f5bea1587b81839f54ad14e:/tmp/c3b039f5cc7b34d22b074337e0795120bbade74d8e0881e1f952551ed557638d:/tmp/9b780c238fec09f08e90fa795abb20770a40e705556dd16eb86948fda4d95b7c:/tmp/657d3026000e51f70581b7ec09085f45090995cbb4156b5f375396ff5410ebcc:/tmp/0748ffb8e4c5a0841bc02d0d95dca7e36d913e20b405a0110665ff927123665b:/tmp/049697104d295f71dbd35643a4d192c0cb9aa781d17d558f82639b123e32572c:/tmp/e55677c04935443363320fea8a96ff46179d7ac01e808b9e0a9b0c5461b4fded:/tmp/d0dc7843a71f2c4e3ec55a72787b7ab01289b97d711cb376f442f306c514ae79:/tmp/44467e4bf1377f6bbdac2521e36f450c9076f59f3f47dc365a98c0bf94319451:/tmp/25a556a92ebe834674bf1be93a27559b2c842ef85207e8277378911abe800c6e:/tmp/b27320c9edea679af71e97c2c27d4c9ec6b55e66ca129a58f9d5b819cee4a794:/tmp/68310bbebb145246a72ac691c5215760f540974fb26a955963a99899ebf4a224:/tmp/8c44e31e47470672ab7cd8f019fc1c593089e9562f24222c6b7d1006458ca949:/tmp/2f6507b01201917261fab6b1b32cbef01ec37174497a2eabb0e62da9d5b7e8ab:/tmp/9b990bfc7f2b9ba92278f278a167be25234cbced2b19cd3fbbcced5302839efe:/tmp/734e90b481e8fdd2da47d07f3ad7afbc6cd9b9151c2f67599920803d805a7b29:/tmp/a8253cca3540f9e76a48c05b4671ccd91617e9f604b25ace22f483cae504157e:/tmp/a1b00e00886530fbd2b3d10c2debafe02be083b35da394a8818d129bac9a27a8:/tmp/8e983c37b0676ba63650b8aaa2266493a6c03dcd7e741169318b06670c9176d4:/tmp/67248fa878d70e85c59e3018b129383e130dbe5984f73cf65b18879e380adceb:/tmp/f8b42faebe52c312b5b0c76dcbfafb8f3cab5f6742cc430e6add02b7e86f34d1:/tmp/396fab63cd4baf815dd559c83275aa1cd7254b22005a939cd3eb943ce37dbaff:/tmp/4d8a4b6c43ccb2549385b6456e9945209911fa2b45c7301f79f810a91781b8e4:/tmp/db02eff6417ce29a8fa8d5df8c5bc0fb2db9c763a7350952db777f6b3e9f8456:/tmp/610b6e7c27d57847da697fc26f0ceffdb8e66d6fd8ab78b377db521008b361f9:/tmp/3b1e8147b693b273c044d31a194524f6aefc1dd6708ac6a1c00cdd817fcb0004:/tmp/d1f370bc1f5db4830e585a05dbd005e4f5d2f27f7c240c9f486ddb0d7eb6954c:/tmp/46d7abc3697064e1f7018e5fb5b2085571e5fbe74332570635a6e571966c7c29:/tmp/23e068193b591383271b78eec92998114e811111eb933fd5808302631ce32843:/tmp/e7cec04e351a1976ca396c5bcfbf60b21565012c66ba8dec053cfdcf1a158bc6:/tmp/00bb41e122a67e8d98795522a141ff586b0d71737f33104c621563cd6843afad:/tmp/27a8ae212bee8b1abdff4eba0e7fe8ea1d572efcf1bc1ac93d75f2a9308c0809:/tmp/70bcb1394a8f7d6667929b4a1bdc57ffd54d8a8a423e391e0ce6d9c8b205f86e:/tmp/d88fbf9b75e88420304344114086a3a151c89a0224d1d0ec6e28edea840f2c60:/tmp/416def6231e4b02156867f939ac9f2fd5ddc211f6dcab54feb66094754152db2:/tmp/8aa5294cb21aab7c28a346fac88b90490ef567c28c96604f045e7855bcf86018:/tmp/510de218389e5787030c46436840b85a7669a971cd8b7893d871076d05bcb440:/tmp/c312c8301aa15f06f087f62b6d50e5f682bb49d1c1c85f4af0572532ae4d3135:/tmp/06136625ac03b1f3fbe3ed16fac5d763a0237719265965108432d38b6f43af11:/tmp/48266d4870e9a65a2196312a1df7df6d54834e1901e83ee7a89cf6d15c3d6aef:/tmp/ce31a3bbe0cfd6ba58c04d897d6475d166908b78231b3432092011bb62d3da36:/tmp/d6ffd30a4becac380a7525f67e1843e9f781f1f70c58c7c204d7e9f1c4002a83:/tmp/d9c872d1b12db539acd24ea950392d93858ee4992220eb60d10e6474fc08161d:/tmp/a340b040978521cb0df4ce2b1789e52d08781d568e915dff07abd8e73d063781:/tmp/82c2dadae70bd0e14da45c34bf23d92b7d167579860f2fac690b9de7a9e72890:/tmp/ff9373e51335733a04219a47e32b8013d437409efb46933ec1364c4b21ce2296:/tmp/239e42b6555538224b25f90184da2e1231471caac31fdd057d949188123af61b:/tmp/39e11597839e77447636789025fd9c028f297414ad880003d60108449cfc61dc:/tmp/36b6a2ccc55d35201c4428d7c38b0dee5de7aabd6d41ac0869c1c082cb99f594,upperdir=/tmp/f70353ec17f61ed4ecb87904cb38cf32091d6deebb9833ee2fda58a3ce681442/upper,workdir=/tmp/f70353ec17f61ed4ecb87904cb38cf32091d6deebb9833ee2fda58a3ce681442/work /tmp/f70353ec17f61ed4ecb87904cb38cf32091d6deebb9833ee2fda58a3ce681442-mount failed with exit code 255
```

I think the cause is the length of the command, which is over 4096. Probably, this is the limit for command line arguments and the command cannot be executed correctly.

**Steps to reproduce the issue:**
Run the follwing Dockerfile with: `docker build --platform=linux .`
```Dockerfile
FROM alpine as first
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test
RUN echo test > /test

FROM alpine
COPY --from=first /test /test
```

**Output of `docker version`:**

```
Client:
 Version:       18.03.0-ce
 API version:   1.37
 Go version:    go1.9.4
 Git commit:    0520e24
 Built: Wed Mar 21 23:06:28 2018
 OS/Arch:       windows/amd64
 Experimental:  false
 Orchestrator:  swarm

Server:
 Engine:
  Version:      18.03.0-ce
  API version:  1.37 (minimum version 1.24)
  Go version:   go1.9.4
  Git commit:   0520e24
  Built:        Wed Mar 21 23:21:06 2018
  OS/Arch:      windows/amd64
  Experimental: true
```

**Output of `docker info`:**

```
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 123
Server Version: 18.03.0-ce
Storage Driver: windowsfilter (windows) lcow (linux)
 Windows:
 LCOW:
Logging Driver: json-file
Plugins:
 Volume: local
 Network: ics l2bridge l2tunnel nat null overlay transparent
 Log: awslogs etwlogs fluentd gelf json-file logentries splunk syslog
Swarm: inactive
Default Isolation: hyperv
Kernel Version: 10.0 16299 (16299.15.amd64fre.rs3_release.170928-1534)
Operating System: Windows 10 Enterprise
OSType: windows
Architecture: x86_64
CPUs: 8
Total Memory: 15.89GiB
Name: YUSUFG-PC
ID: 7BCW:GTLJ:2FQB:KV3J:NRRE:IHTW:6C3B:YOCE:65GK:NKHJ:IS3D:3LNJ
Docker Root Dir: C:\ProgramData\Docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Labels:
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Windows 10 1709 physical machine.

### PR DESCRIPTION
**Pull request by lowenna**
Signed-off-by: John Howard <jhoward@microsoft.com>
    
Fixes #36764. @johnstep PTAL. @jterry75 FYI.
    
There are two commits in this PR. The first ensure that errors are actually returned to the caller - it was being thrown away.
    
The second commit changes the LCOW driver to map, on a per service VM basis, "long" container paths such as `/tmp/c8fa0ae1b348f505df2707060f6a49e63280d71b83b7936935c827e2e9bde16d` to much shorter paths, based on a per-service VM counter, so something more like /tmp/d3. This means that the root cause of the failure where the mount call to create the overlay was failing due to command line length becomes something much shorter such as below.
    
`mount -t overlay overlay -olowerdir=/tmp/d3:/tmp/d4:/tmp/d5:/tmp/d6:/tmp/d7:/tmp/d8:/tmp/d9:/tmp/d10:/tmp/d11:/tmp/d12:/tmp/d13:/tmp/d14:/tmp/d15:/tmp/d16:/tmp/d17:/tmp/d18:/tmp/d19:/tmp/d20:/tmp/d21:/tmp/d22:/tmp/d23:/tmp/d24:/tmp/d25:/tmp/d26:/tmp/d27:/tmp/d28:/tmp/d29:/tmp/d30:/tmp/d31:/tmp/d32:/tmp/d33:/tmp/d34:/tmp/d35:/tmp/d36:/tmp/d37:/tmp/d38:/tmp/d39:/tmp/d40:/tmp/d41:/tmp/d42:/tmp/d43:/tmp/d44:/tmp/d45:/tmp/d46:/tmp/d47:/tmp/d48:/tmp/d49:/tmp/d50:/tmp/d51:/tmp/d52:/tmp/d53:/tmp/d54:/tmp/d55:/tmp/d56:/tmp/d57:/tmp/d58:/tmp/d59:/tmp/d60:/tmp/d61:/tmp/d62,upperdir=/tmp/d2/upper,workdir=/tmp/d2/work /tmp/c8fa0ae1b348f505df2707060f6a49e63280d71b83b7936935c827e2e9bde16d-mount`
    
For those worrying about overflow (which I'm sure @thaJeztah will mention...): It's safe to use a counter here as SVMs are disposable in the default configuration. The exception is when running the daemon in unsafe LCOW "global" mode (ie `--storage-opt lcow.globalmode=1`) where the SVMs aren't disposed of, but a single one is reused. However, to overflow the command line length, it would require several hundred-thousand trillion (conservative, I should sit down and work it out accurately if I get -really- bored) of SCSI hot-add operations, and even to hit that would be hard as just running containers normally uses the VPMEM path for the containers UVM, not to the global SVM on SCSI. It gets incremented by one per build step (commit more accurately) as a general rule. Hence it would be necessary to have to be doing automated builds without restarting the daemon for literally years on end in unsafe mode. :innocent:
    
Note that in reality, the previous limit of ~47 layers before hitting the command line length limit is close to what is possible in the platform, at least as of RS5/Windows Server 2019 where, in the HCS v1 schema, a single SCSI controller is used, and that can only support 64 disks per controller per the Hyper-V VDEV. And remember we have one slot taken up for the SVMs scratch, and another for the containers scratch when committing a layer. So the best you can architecturally get on the platform in global mode is the following (it's +1 in default mode so you could have one extra `RUN echo test > /test`)
    
```
    PS E:\docker\build\36764\short> docker build --no-cache .
    Sending build context to Docker daemon  2.048kB
    Step 1/4 : FROM alpine as first
     ---> 11cd0b38bc3c
    Step 2/4 : RUN echo test > /test
     ---> Running in 8ddfe20e5bfb
    Removing intermediate container 8ddfe20e5bfb
     ---> b0103a00b1c9
    Step 3/4 : FROM alpine
     ---> 11cd0b38bc3c
    Step 4/4 : COPY --from=first /test /test
     ---> 54bfae391eba
    Successfully built 54bfae391eba
    PS E:\docker\build\36764\short> cd ..
    PS E:\docker\build\36764> docker build --no-cache .
    Sending build context to Docker daemon  4.689MB
    Step 1/61 : FROM alpine as first
     ---> 11cd0b38bc3c
    Step 2/61 : RUN echo test > /test
     ---> Running in 02597ff870db
    Removing intermediate container 02597ff870db
     ---> 3096de6fc454
    Step 3/61 : RUN echo test > /test
     ---> Running in 9a8110f4ff19
    Removing intermediate container 9a8110f4ff19
     ---> 7691808cf28e
    Step 4/61 : RUN echo test > /test
     ---> Running in 9afb8f51510b
    Removing intermediate container 9afb8f51510b
     ---> e42a0df2bb1c
    Step 5/61 : RUN echo test > /test
     ---> Running in fe977ed6804e
    Removing intermediate container fe977ed6804e
     ---> 55850c9b0479
    Step 6/61 : RUN echo test > /test
     ---> Running in be65cbfad172
    Removing intermediate container be65cbfad172
     ---> 0cf8acba70f0
    Step 7/61 : RUN echo test > /test
     ---> Running in fd5b0907b6a9
    Removing intermediate container fd5b0907b6a9
     ---> 257a4493d85d
    Step 8/61 : RUN echo test > /test
     ---> Running in f7ca0ffd9076
    Removing intermediate container f7ca0ffd9076
     ---> 3baa6f4fa2d5
    Step 9/61 : RUN echo test > /test
     ---> Running in 5146814d4727
    Removing intermediate container 5146814d4727
     ---> 485b9d5cf228
    Step 10/61 : RUN echo test > /test
     ---> Running in a090eec1b743
    Removing intermediate container a090eec1b743
     ---> a7eb10155b51
    Step 11/61 : RUN echo test > /test
     ---> Running in 942660b288df
    Removing intermediate container 942660b288df
     ---> 9d286a1e2133
    Step 12/61 : RUN echo test > /test
     ---> Running in c3d369aa91df
    Removing intermediate container c3d369aa91df
     ---> f78be4788992
    Step 13/61 : RUN echo test > /test
     ---> Running in a03c3ac6888f
    Removing intermediate container a03c3ac6888f
     ---> 6504363f61ab
    Step 14/61 : RUN echo test > /test
     ---> Running in 0c3c2fca3f90
    Removing intermediate container 0c3c2fca3f90
     ---> fe3448b8bb29
    Step 15/61 : RUN echo test > /test
     ---> Running in 828d51c76d3b
    Removing intermediate container 828d51c76d3b
     ---> 870684e3aea0
    Step 16/61 : RUN echo test > /test
     ---> Running in 59a2f7c5f3ad
    Removing intermediate container 59a2f7c5f3ad
     ---> cf84556ca5c0
    Step 17/61 : RUN echo test > /test
     ---> Running in bfb4e088eeb3
    Removing intermediate container bfb4e088eeb3
     ---> 9c8f9f652cef
    Step 18/61 : RUN echo test > /test
     ---> Running in f1b88bb5a2d7
    Removing intermediate container f1b88bb5a2d7
     ---> a6233ad21648
    Step 19/61 : RUN echo test > /test
     ---> Running in 45f70577d709
    Removing intermediate container 45f70577d709
     ---> 1b5cc52d370d
    Step 20/61 : RUN echo test > /test
     ---> Running in 2ce231d5043d
    Removing intermediate container 2ce231d5043d
     ---> 4a0e17cbebaa
    Step 21/61 : RUN echo test > /test
     ---> Running in 52e4b0928f1f
    Removing intermediate container 52e4b0928f1f
     ---> 99b50e989bcb
    Step 22/61 : RUN echo test > /test
     ---> Running in f7ba3da7460d
    Removing intermediate container f7ba3da7460d
     ---> bfa3cad88285
    Step 23/61 : RUN echo test > /test
     ---> Running in 60180bf60f88
    Removing intermediate container 60180bf60f88
     ---> fe7271988bcb
    Step 24/61 : RUN echo test > /test
     ---> Running in 20324d396531
    Removing intermediate container 20324d396531
     ---> e930bc039128
    Step 25/61 : RUN echo test > /test
     ---> Running in b3ac70fd4404
    Removing intermediate container b3ac70fd4404
     ---> 39d0a11ea6d8
    Step 26/61 : RUN echo test > /test
     ---> Running in 0193267d3787
    Removing intermediate container 0193267d3787
     ---> 8062d7aab0a5
    Step 27/61 : RUN echo test > /test
     ---> Running in f41f45fb7985
    Removing intermediate container f41f45fb7985
     ---> 1f5f18f2315b
    Step 28/61 : RUN echo test > /test
     ---> Running in 90dd09c63d6e
    Removing intermediate container 90dd09c63d6e
     ---> 02f0a1141f11
    Step 29/61 : RUN echo test > /test
     ---> Running in c557e5386e0a
    Removing intermediate container c557e5386e0a
     ---> dbcd6fb1f6f4
    Step 30/61 : RUN echo test > /test
     ---> Running in 65369385d855
    Removing intermediate container 65369385d855
     ---> e6e9058a0650
    Step 31/61 : RUN echo test > /test
     ---> Running in d861fcc388fd
    Removing intermediate container d861fcc388fd
     ---> 6e4c2c0f741f
    Step 32/61 : RUN echo test > /test
     ---> Running in 1483962b7e1c
    Removing intermediate container 1483962b7e1c
     ---> cf8f142aa055
    Step 33/61 : RUN echo test > /test
     ---> Running in 5868934816c1
    Removing intermediate container 5868934816c1
     ---> d5ff87cdc204
    Step 34/61 : RUN echo test > /test
     ---> Running in e057f3201f3a
    Removing intermediate container e057f3201f3a
     ---> b4031b7ab4ac
    Step 35/61 : RUN echo test > /test
     ---> Running in 22b769b9079c
    Removing intermediate container 22b769b9079c
     ---> 019d898510b6
    Step 36/61 : RUN echo test > /test
     ---> Running in f1d364ef4ff8
    Removing intermediate container f1d364ef4ff8
     ---> 9525cafdf04d
    Step 37/61 : RUN echo test > /test
     ---> Running in 5bf505b8bdcc
    Removing intermediate container 5bf505b8bdcc
     ---> cd5002b33bfd
    Step 38/61 : RUN echo test > /test
     ---> Running in be24a921945c
    Removing intermediate container be24a921945c
     ---> 8675db44d1b7
    Step 39/61 : RUN echo test > /test
     ---> Running in 352dc6beef3d
    Removing intermediate container 352dc6beef3d
     ---> 0ab0ece43c71
    Step 40/61 : RUN echo test > /test
     ---> Running in eebde33e5d9b
    Removing intermediate container eebde33e5d9b
     ---> 46ca4b0dfc03
    Step 41/61 : RUN echo test > /test
     ---> Running in f920313a1e85
    Removing intermediate container f920313a1e85
     ---> 7f3888414d58
    Step 42/61 : RUN echo test > /test
     ---> Running in 10e2f4dc1ac7
    Removing intermediate container 10e2f4dc1ac7
     ---> 14db9e15f2dc
    Step 43/61 : RUN echo test > /test
     ---> Running in c849d6e89aa5
    Removing intermediate container c849d6e89aa5
     ---> fdb770494dd6
    Step 44/61 : RUN echo test > /test
     ---> Running in 419d1a8353db
    Removing intermediate container 419d1a8353db
     ---> d12e9cf078be
    Step 45/61 : RUN echo test > /test
     ---> Running in 0f1805263e4c
    Removing intermediate container 0f1805263e4c
     ---> cd005e7b08a4
    Step 46/61 : RUN echo test > /test
     ---> Running in 5bde05b46441
    Removing intermediate container 5bde05b46441
     ---> 05aa426a3d4a
    Step 47/61 : RUN echo test > /test
     ---> Running in 01ebc84bd1bc
    Removing intermediate container 01ebc84bd1bc
     ---> 35d371fa4342
    Step 48/61 : RUN echo test > /test
     ---> Running in 49f6c2f51dd4
    Removing intermediate container 49f6c2f51dd4
     ---> 1090b5dfa130
    Step 49/61 : RUN echo test > /test
     ---> Running in f8a9089cd725
    Removing intermediate container f8a9089cd725
     ---> b2d0eec0716d
    Step 50/61 : RUN echo test > /test
     ---> Running in a1697a0b2db0
    Removing intermediate container a1697a0b2db0
     ---> 10d96ac8f497
    Step 51/61 : RUN echo test > /test
     ---> Running in 33a2332c06eb
    Removing intermediate container 33a2332c06eb
     ---> ba5bf5609c1c
    Step 52/61 : RUN echo test > /test
     ---> Running in e8920392be0d
    Removing intermediate container e8920392be0d
     ---> 5b3a95685c7e
    Step 53/61 : RUN echo test > /test
     ---> Running in 4b9298587c65
    Removing intermediate container 4b9298587c65
     ---> d4961a349141
    Step 54/61 : RUN echo test > /test
     ---> Running in 8a0c960c2ba1
    Removing intermediate container 8a0c960c2ba1
     ---> b413197fcfa2
    Step 55/61 : RUN echo test > /test
     ---> Running in 536ee3b9596b
    Removing intermediate container 536ee3b9596b
     ---> fc16b69b224a
    Step 56/61 : RUN echo test > /test
     ---> Running in 8b817b8d7b59
    Removing intermediate container 8b817b8d7b59
     ---> 2f0896400ff9
    Step 57/61 : RUN echo test > /test
     ---> Running in ab0ed79ec3d4
    Removing intermediate container ab0ed79ec3d4
     ---> b4fb420e736c
    Step 58/61 : RUN echo test > /test
     ---> Running in 8548d7eead1f
    Removing intermediate container 8548d7eead1f
     ---> 745103fd5a38
    Step 59/61 : RUN echo test > /test
     ---> Running in 1980559ad5d6
    Removing intermediate container 1980559ad5d6
     ---> 08c1c74a5618
    Step 60/61 : FROM alpine
     ---> 11cd0b38bc3c
    Step 61/61 : COPY --from=first /test /test
     ---> 67f053c66c27
    Successfully built 67f053c66c27
    PS E:\docker\build\36764>
```
    
Note also that subsequent error messages once you go beyond current platform limitations kind of suck (such as insufficient resources with a bunch of spew which is incomprehensible to most) and we could do better to detect this earlier in the daemon. That'll be for a (reasonably low-priority) follow-up though as and when I have time. Theoretically we *may*, if the platform doesn't require additional changes for RS5, be able to have bigger platform limits using the v2 schema with up to 127 VPMem devices, and the possibility to have multiple SCSI controllers per SVM/UVM. However, currently LCOW is using HCS v1 schema calls, and there's no plans to rewrite the graphdriver/libcontainerd components outside of the moving LCOW fully over to the containerd runtime/snapshotter using HCS v2 schema, which is still some time off fruition.
    
PS OK, while waiting for a full run to complete, I did get bored. Turns out it won't overflow line length as max(uint64) is 18446744073709551616 which would still be short enough at 127 layers, double the current platform limit. And I could always change it to hex or base36 to make it even shorter, or remove the 'd' from /tmp/dN. IOW, pretty sure no-one is going to hit the limit even if we could get the platform to 256 which is the current Hyper-V SCSI limit per VM (4x64), although PMEM at 127 would be the next immediate limit.

### CODE REVIEW

**Comment by jhowardmsft** [comment]
Latest janky and z CI failures are unrelated. Restarted.

File: builder/remotecontext/detect.go [hunk 1]
```diff
@@ -173,6 +174,9 @@ func StatAt(remote builder.Source, path string) (os.FileInfo, error) {
 func FullPath(remote builder.Source, path string) (string, error) {
 	fullPath, err := remote.Root().ResolveScopedPath(path, true)
 	if err != nil {
+		if runtime.GOOS == "windows" {
+			return "", fmt.Errorf("failed to resolve scoped path %s (%s): %s. Possible cause is a forbidden path outside the build context", path, fullPath, err)
```
Comment (thaJeztah) [inline builder/remotecontext/detect.go]: Perhaps use `errors.Wrapf(err, "failed to .....")`

**Review by thaJeztah — APPROVED** [review]
SGTM; some notes inline

**Comment by thaJeztah** [comment]
ping @johnstep @kolyshkin PTAL

File: builder/remotecontext/detect.go (hunk 1 above)
Comment (jhowardmsft) [reply builder/remotecontext/detect.go]: I think that would make it non-sensicle and put the actual error in the wrong place.

File: daemon/graphdriver/lcow/lcow.go [hunk 2]
```diff
@@ -212,6 +212,17 @@ func (d *Driver) getVMID(id string) string {
 	return id
 }
 
+// remapLongToShortContainerPath does the mapping of a long container path for a
+// SCSI attached disk, to a short container path where it's actually mounted.
+func remapLongToShortContainerPath(longContainerPath string, attachCounter uint64, svmName string) string {
+	shortContainerPath := longContainerPath
+	if shortContainerPath != "" && shortContainerPath != toolsScratchPath {
+		shortContainerPath = fmt.Sprintf("/tmp/d%d", attachCounter)
+		logrus.Debugf("lcowdriver: UVM %s: remapping %s --> %s", svmName, longContainerPath, shortContainerPath)
```
[the remark this answers is not in the document]
Comment (jhowardmsft) [reply daemon/graphdriver/lcow/lcow.go]: Sure, but it's consistent currently. For a followon perhaps

**Review by johnstep — APPROVED** [review]

File: daemon/graphdriver/lcow/lcow.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (thaJeztah) [reply daemon/graphdriver/lcow/lcow.go]: yes; definitely

File: daemon/graphdriver/lcow/lcow_svm.go [hunk 3]
```diff
@@ -385,7 +386,15 @@ func (svm *serviceVM) deleteUnionMount(mountName string, disks ...hcsshim.Mapped
...
-	process, err := svm.config.RunProcess(command, stdin, stdout, stderr)
+	var process hcsshim.Process
+	var err error
+	errOut := &bytes.Buffer{}
+
+	if stderr != nil {
+		process, err = svm.config.RunProcess(command, stdin, stdout, stderr)
+	} else {
+		process, err = svm.config.RunProcess(command, stdin, stdout, errOut)
+	}
```
[the remark this answers is not in the document]
Comment (thaJeztah) [reply daemon/graphdriver/lcow/lcow_svm.go]: @johnstep did you want that in this PR, or in a follow up? (CI looks green, so if a follow up; feel free to merge)

[the remark this answers is not in the document]
Comment (johnstep) [reply daemon/graphdriver/lcow/lcow_svm.go]: Just a nit; a follow-up is fine.

**Pull request merged**